torchgeo/trainers/utils.py

```diff
@@ -95,3 +97,51 @@ def load_state_dict(model: Module, state_dict: Dict[str, Tensor]) -> Module:
     model.load_state_dict(state_dict, strict=False)  # type: ignore[arg-type]
 
     return model
+
+
+def reinit_initial_conv_layer(
+    layer: Conv2d, new_in_channels: int, keep_rgb_weights: bool
+) -> Conv2d:
+    """Clones a Conv2d layer while optionally retaining some of the original weights.
+
+    When replacing the first convolutional layer in a model with one that operates over
+    different number of input channels, we sometimes want to keep a subset of the kernel
+    weights the same (e.g. the RGB weights of an ImageNet pretrained model). This is a
+    convinience function that performs that function.
+
+    Args:
+        layer: the Conv2d layer to initialize
+        new_in_channels: the new number of input
+        keep_rgb_weights: flag indicating whether to re-initialize the first 3 channels
+
+    Returns:
+        a Conv2d layer with new kernel weights
+    """
+    use_bias = layer.bias is not None
+    if keep_rgb_weights:
+        w_old = layer.weight.data[:, :3, :, :].clone()
+        if use_bias:
+            # mypy doesn't realize that bias isn't None here...
+            b_old = layer.bias.data.clone()  # type: ignore[union-attr]
+
+    new_layer = Conv2d(
+        new_in_channels,
+        layer.out_channels,
+        kernel_size=layer.kernel_size,  # type: ignore[arg-type]
+        stride=layer.stride,  # type: ignore[arg-type]
+        padding=layer.padding,  # type: ignore[arg-type]
```

Comment:
For So2Sat we change the stride and padding, right? Maybe these should be parameters of the `reinit_initial_conv_layer` function.

Reply:
Yeah good point!